src/lib/MrDocsCompilationDatabase.cpp

```diff
@@ -492,6 +487,18 @@ adjustCommandLine(
       new_cmdline.emplace_back(std::format("-I{}", inc));
     }
 
+    // ------------------------------------------------------
+    // Get driver mode
+    // ------------------------------------------------------
+    // The driver mode distinguishes between clang/gcc and msvc
+    // command line option formats. The value is deduced from
+    // the `-drive-mode` option or from `progName`.
+    // Common values are "gcc", "g++", "cpp", "cl" and "flang".
+    StringRef const driver_mode = driver::getDriverMode(progName, cmdLineCStrs);
+    // Identify if we should use "msvc/clang-cl" or "clang/gcc" format
+    // for options.
+    bool const is_clang_cl = driver::IsClangCL(driver_mode);
+
```

Comment:
This logic being moved down here is also great. It limits errors that could go up there.